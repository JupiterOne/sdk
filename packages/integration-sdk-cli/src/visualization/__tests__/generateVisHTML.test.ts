import type { Edge, Node } from 'vis';

import {
  generateVisHTML,
  nothingToDisplayMessage,
} from '../../utils/generateVisHTML';

const nodeDataSets: Node[] = [
  { id: '1', label: 'Node 1' },
  { id: '2', label: 'Node 2' },
  { id: '3', label: 'Node 3' },
];

const edgeDataSet: Edge[] = [
  { from: '1', to: '2' },
  { from: '2', to: '3' },
  { from: '3', to: '1' },
];

test('renders html with default config', () => {
  const html = generateVisHTML('path/to/data', nodeDataSets, edgeDataSet);

  expect(html).toContain(
    'var options = {"edges":{"arrows":{"to":{"enabled":true}}},"physics":{"barnesHut":{"springLength":300,"centralGravity":0.03}}}',
  );
  expect(html).toMatchSnapshot();
});

test('renders custom config when passed in', () => {
  const html = generateVisHTML('path/to/data', nodeDataSets, edgeDataSet, {
    edges: { color: '#ffffff' },
  });

  expect(html).toContain('"color":"#ffffff"');
  expect(html).toMatchSnapshot();
});

test('renders nothing to display message when there is nothing to display', () => {
  const html = generateVisHTML('path/to/data', [], []);

  expect(html).toContain(nothingToDisplayMessage);
  expect(html).toMatchSnapshot();
});
