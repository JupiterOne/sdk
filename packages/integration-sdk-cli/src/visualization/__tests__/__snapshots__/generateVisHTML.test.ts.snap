// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders custom config when passed in 1`] = `
"<!DOCTYPE html>
    <html lang=\\"en\\">
    <head>
        <meta charset=\\"UTF-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
        <title>Integration SDK Graph</title>
    </head>
    <body style=\\"height:100vh;width:100%\\">
      <div id=\\"integration-graph\\" style=\\"height:100vh;width:100%\\"></div>
      <script type=\\"text/javascript\\" src=\\"https://unpkg.com/vis-network@7.6.1/standalone/umd/vis-network.min.js\\"></script>
      <script type=\\"text/javascript\\">
        var nodes = new vis.DataSet([{\\"id\\":\\"1\\",\\"label\\":\\"Node 1\\"},{\\"id\\":\\"2\\",\\"label\\":\\"Node 2\\"},{\\"id\\":\\"3\\",\\"label\\":\\"Node 3\\"}]);
        var edges = new vis.DataSet([{\\"from\\":\\"1\\",\\"to\\":\\"2\\"},{\\"from\\":\\"2\\",\\"to\\":\\"3\\"},{\\"from\\":\\"3\\",\\"to\\":\\"1\\"}]);
        // create a network
        var container = document.getElementById(\\"integration-graph\\");
        var data = {
          nodes: nodes,
          edges: edges
        };
        var options = {\\"edges\\":{\\"color\\":\\"#ffffff\\",\\"arrows\\":{\\"to\\":{\\"enabled\\":true}}},\\"physics\\":{\\"barnesHut\\":{\\"springLength\\":300,\\"centralGravity\\":0.03}}};
        var network = new vis.Network(container, data, options);
      </script>
    </body>
    </html>"
`;

exports[`renders html with default config 1`] = `
"<!DOCTYPE html>
    <html lang=\\"en\\">
    <head>
        <meta charset=\\"UTF-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
        <title>Integration SDK Graph</title>
    </head>
    <body style=\\"height:100vh;width:100%\\">
      <div id=\\"integration-graph\\" style=\\"height:100vh;width:100%\\"></div>
      <script type=\\"text/javascript\\" src=\\"https://unpkg.com/vis-network@7.6.1/standalone/umd/vis-network.min.js\\"></script>
      <script type=\\"text/javascript\\">
        var nodes = new vis.DataSet([{\\"id\\":\\"1\\",\\"label\\":\\"Node 1\\"},{\\"id\\":\\"2\\",\\"label\\":\\"Node 2\\"},{\\"id\\":\\"3\\",\\"label\\":\\"Node 3\\"}]);
        var edges = new vis.DataSet([{\\"from\\":\\"1\\",\\"to\\":\\"2\\"},{\\"from\\":\\"2\\",\\"to\\":\\"3\\"},{\\"from\\":\\"3\\",\\"to\\":\\"1\\"}]);
        // create a network
        var container = document.getElementById(\\"integration-graph\\");
        var data = {
          nodes: nodes,
          edges: edges
        };
        var options = {\\"edges\\":{\\"arrows\\":{\\"to\\":{\\"enabled\\":true}}},\\"physics\\":{\\"barnesHut\\":{\\"springLength\\":300,\\"centralGravity\\":0.03}}};
        var network = new vis.Network(container, data, options);
      </script>
    </body>
    </html>"
`;

exports[`renders nothing to display message when there is nothing to display 1`] = `
"<!DOCTYPE html>
    <html lang=\\"en\\">
    <head>
        <meta charset=\\"UTF-8\\">
        <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\">
        <title>Integration SDK Graph</title>
    </head>
    <body style=\\"height:100vh;width:100%\\">
      <h1 style=\\"text-align: center;\\">path/to/data: There was no data found to visualize.</h1>
    </body>
    </html>"
`;
